stdlib/builtins.pyi

```diff
@@ -1619,12 +1619,7 @@ class reversed(Iterator[_T], Generic[_T]):
     def __length_hint__(self) -> int: ...
 
 def repr(__obj: object) -> str: ...
-@overload
-def round(number: SupportsRound[Any]) -> int: ...
-@overload
-def round(number: SupportsRound[Any], ndigits: None) -> int: ...
-@overload
-def round(number: SupportsRound[_T], ndigits: SupportsIndex) -> _T: ...
+def round(number: SupportsRound[_T], ndigits: SupportsIndex | None = ...) -> _T: ...
```

Comment:
This needs to be an overloaded method so that mypy can infer that `round(3.14) -> int`, but `round(3.14, ndigits=1) -> float`. That's causing the vast majority of the primer hits.

Reply:
Again, first arg can be like this:

```python
>>> class R:
...       def __round__(self): 
...           return 'a'
... 
>>> round(R())
'a'
```

Reply:
The implementation does indeed just return whatever `__round__` returns. Maybe we need two protocols, one for zero-argument `__round__` and one for one-argument  `__round__`.

Reply:
Just to be clear, two argument `round` is the same: it can return anything.

```python
>>> class R2:
...    def __round__(self, n):
...       return 'a'
... 
>>> round(R2(), 5)
'a'
```